src/python/WMComponent/WorkflowUpdater/SiteListPoller.py

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env python
+"""
+File       : SiteListPoller
+Author     : Valentin Kuznetsov <vkuznet AT gmail dot com>
+Description: module to update of site lists within a WMAgent
+"""
+
+# system modules
+import json
+import shutil
+import logging
+import threading
+
+# WMCore modules
+from Utils.Timers import timeFunction
+from WMCore.DAOFactory import DAOFactory
+from WMCore.Services.WorkQueue.WorkQueue import WorkQueue
+from WMCore.Services.WMStatsServer.WMStatsServer import WMStatsServer
+from WMCore.WMSpec.WMWorkload import WMWorkloadHelper
+from WMCore.WorkerThreads.BaseWorkerThread import BaseWorkerThread
+
+
+class SiteListPoller(BaseWorkerThread):
+    def __init__(self, config):
+        """
+        Initialize SiteListPoller object
+        :param config: a Configuration object with the component configuration
+        """
+        BaseWorkerThread.__init__(self)
+        myThread = threading.currentThread()
+        self.logger = myThread.logger
+
+        # the reqmgr2Url should be points to ReqMgr2 data services, i.e. /reqmgr2 end-point
+        self.wmstatsUrl = getattr(config.SiteListPoller, "wmstatsUrl")
+        self.wmstatsSrv = WMStatsServer(config.wmstatsSvc_url, logger=self.logger)
+
+        # provide access to WMBS in local WMAgent
+        self.daoFactory = DAOFactory(package="WMCore.WMBS",
+                                     logger=myThread.logger,
+                                     dbinterface=myThread.dbi)
+        # DB function to retrieve active workflows
+        self.listActiveWflows = self.daoFactory(classname="Workflow.GetUnfinishedWorkflows")
+
+        # local WorkQueue service
+        self.localCouchUrl = self.config.WorkQueueManager.couchurl
+        self.localWQ = WorkQueue(self.localCouchUrl,
+                                 self.config.WorkQueueManager.dbname)
+
+    def getActiveWorkflows(self):
+        """
+        Provide list of active requests within WMAgent
+        :return: dict of workflows names vs pickle files
+        """
+        # get list of active workflows in WMAgent
+        wflowSpecs = self.listActiveWflows.execute()
+
+        # construct dictionary of workflow names and their pickle files
+        wmaSpecs = {}
+        for wflowSpec in wflowSpecs:
+            name = wflowSpec['name']  # this is the name of workflow
+            pklFileName = wflowSpec['spec']  # the "spec" in WMBS table (wmbs_workflow.spec) is pkl file name
+            wmaSpecs[name] = pklFileName
+        return wmaSpecs
+
+    def getRequestSpecs(self, requests):
```

Comment:
The word "spec" seems to be slightly overloaded in this module.
Hence, I would suggest to rename this method with something more meaningful (e.g. getRequestSites)

Reply:
ok, changed.